DH_Weapons/Classes/DH_MolotovProjectile.uc

```diff
@@ -0,0 +1,644 @@
+    if( FluidSurfaceInfo(other)!=none )
+    {
+        CheckForSplash( Location );
+    }
+
+    if( other==none || (!other.bProjTarget && !other.bBlockActors) )
+    {
+        return;
+    }
+
+    // We use TraceThisActor do a simple line check against the actor we've hit, to get an accurate HitLocation to pass to ProcessTouch()
+    // It's more accurate than using our current location as projectile has often travelled further by the time this event gets called
+    // But if that trace returns true then it somehow didn't hit the actor, so we fall back to using our current location as the HitLocation
+    // Also skip trace & use our location if velocity is zero (touching actor when projectile spawns) or hit a Mover actor (legacy, don't know why)
+    if(
+            VSizeSquared(Velocity)==0
+        ||  other.IsA(class'Mover'.Name)
+        ||  other.TraceThisActor(
+                /*out*/ hitLocation ,
+                /*out*/ hitNormal ,
+                Location ,
+                Location-(2.0*Velocity) ,
+                GetCollisionExtent()
+            )
+    )
+    {
+        hitLocation = Location;
+    }
+
+    // Special handling for hit on a collision mesh actor - switch hit actor to CM's owner & proceed as if we'd hit that actor
+    if( other.IsA(class'DHCollisionMeshActor'.Name) )
+    {
+        if( DHCollisionMeshActor(other).bWontStopThrownProjectile )
+        {
+            return; // exit, doing nothing, if col mesh actor is set not to stop a thrown projectile, e.g. grenade or satchel
+        }
+
+        other = other.Owner;
+    }
+
+    ProcessTouch( other , hitLocation );
+}
+
+simulated function ProcessTouch ( Actor other , vector hitLocation )
+{
+    local vector tempHitLocation, hitNormal;
+
+    if(
+        other==Instigator
+        || other.Base==Instigator
+        || ROBulletWhipAttachment(other)!=none
+    )
+    {
+        return;
+    }
+
+    Trace(
+        /*out*/ tempHitLocation ,
+        /*out*/ hitNormal ,
+        hitLocation + Normal(Velocity) * 50.0 ,
+        hitLocation - Normal(Velocity) * 50.0 ,
+        true
+    ); // get a reliable HitNormal for a deflection
+
+    HitWall( hitNormal , other );
+}
+
+/// <summary> Calls Destroy() </summary>
+simulated function Explode ( vector hitLocation , vector hitNormal )
+{
+    if( !bDud ) BlowUp( hitLocation );
+    Destroy();
+}
+
+/// <summary> Deals damage, creates fx & sfx </summary>
+simulated function BlowUp ( vector hitLocation )
+{
+    local Actor instance, tracedActorHit;
+    local ESurfaceTypes hitSurfaceType;
+    local vector tracedHitPoint, tracedHitNormal;
+    local rotator spread;
+    local int i;
+
+    if( Role==ROLE_Authority )
+    {
+        MakeNoise( 1.0 );
+        if( Level.NetMode!=NM_DedicatedServer )
+        {
+            PlaySound( ExplosionSound[Rand(3)] ,, 5.0 ,, ExplosionSoundRadius , 1.0 , true );
+        }
+        
+        if( EffectIsRelevant(Location,false) )
```

Comment:
`Role == ROLE_Authority` and `EffectIsRelevant()` are mutually exclusive in multiplayer. Effects are only relevant on the client, while the the first check ensures that things are executed on the server only (the `BlowUp()` function is never called on the client either).

Reply:
Fixed in commit ebb83f57a1c2f5f2208f013182f4b5c3ee9dd868.